ethcore/src/client/client.rs

```diff
@@ -204,6 +204,9 @@ pub struct Client {
 	/// Database pruning strategy to use for StateDB
 	pruning: journaldb::Algorithm,
 
+	/// Don't prune the state we're currently snapshotting
+	snapshotting_at: RwLock<AtomicU64>,
```

Comment:
you don't need an `RwLock` around atomic

Reply:
I know, but I need to mutate it from a `&self`. What's better, a `Mutex` (but it's written once and read many times)? Or `RwLock<u64>`?

Reply:
Atomics allow modification while having `&self`. This is like an `RwLock<u64>`, but more performant.